Graphics Export

SteffenHeu opened the pull request:
Hey Tomas,
as promised, i started working on the export dialog.
It's not done yet, but this is a first working implementation, so you, @Robinschmid and @Annexhc can comment on the progress.
I made it a module and extended a parameter setup dialog rather than hardcoding everything as before.

Best,
Steffen.

SteffenHeu committed:
minimal working export dialog, now a module

- plots set back to standard size after resizing
- font parameter preview does not update
- master font not set
- sometimes an exception is triggered for some reason

SteffenHeu committed:
Merge branch 'javafx_parameters' into exportfix

tomas-pluskal commented on src/main/java/io/github/mzmine/gui/chartbasics/graphicsexport/GraphicsExportDialogFX.java:
```diff
@@ -0,0 +1,202 @@
+package io.github.mzmine.gui.chartbasics.graphicsexport;
```
Please add the license header to all new files

tomas-pluskal commented on src/main/java/io/github/mzmine/gui/chartbasics/graphicsexport/GraphicsExportDialogFX.java:
```diff
@@ -0,0 +1,202 @@
+      JFreeChart chart) {
+    super(valueCheckRequired, parameterSet);
+
+    chartParam = (ChartThemeParameters) parameterSet
```
No need to cast parameterSet to ChartThemeParameters here - the getParameter() method is universal to all ParameterSet classes

tomas-pluskal commented on src/main/java/io/github/mzmine/gui/chartbasics/graphicsexport/GraphicsExportParameters.java:
```diff
@@ -39,8 +47,9 @@
-  public static final StringParameter path = new StringParameter("Path", "The file path");
-  public static final StringParameter filename = new StringParameter("Filename", "The file name");
+  public static final DirectoryParameter path = new DirectoryParameter("Path", "The file path");
+  public static final StringParameter filename =
```
Wouldn't it make more sense to use the FileParameter instead of separate path + filename?

SteffenHeu replied:
Which FileParameter are you referring to? For the FileNameParameter it would always say the file does not exist.

However, I think it might be easier when exporting multiple graphs to keep the directory loaded and just changing the filename as a string, so you don't have to open the dialog all the time.

tomas-pluskal replied:
Please use the FileNameParameter. The same is used in other export modules like CSV export etc. If you want to have the option to remember the last used directory, it is better to add this feature to the FileNameParameter so all export modules can take advantage of it.

tomas-pluskal commented:
Thanks @SteffenHeu, looks pretty great already!

SteffenHeu committed:
added license, fixed font preview not updating

tomas-pluskal commented on src/main/java/io/github/mzmine/gui/chartbasics/graphicsexport/GraphicsExportModule.java:
```diff
@@ -0,0 +1,48 @@
+package io.github.mzmine.gui.chartbasics.graphicsexport;
```
Please add the license header to all new files

SteffenHeu committed:
added missing license

SteffenHeu committed:
FileNameParameter can now save files

removed basic constructor, every FileNameParameter has to specifiy if it
wants to open or save files

tomas-pluskal commented:
Thanks @SteffenHeu !
